java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -2562,6 +2670,16 @@ public CompletableFuture<Map<String, Map<String, String[][]>>> xreadgroup(
         return commandManager.submitNewCommand(XReadGroup, arguments, this::handleXReadResponse);
     }
 
+    @Override
+    public CompletableFuture<Map<GlideString, Map<GlideString, GlideString[][]>>> xreadgroup(
+            @NonNull Map<GlideString, GlideString> keysAndIds,
+            @NonNull GlideString group,
+            @NonNull GlideString consumer,
+            @NonNull StreamReadGroupOptions options) {
+        GlideString[] arguments = options.toArgsBinary(group, consumer, keysAndIds);
```

Comment:
Cannot use the `ArgsBuilder `because the `toArgs()` function receives a map argument and parses it in a not trivial way (first adds all the keys and than adds all the values).